### PR TITLE
Updating specTemplate and mfRPMRules to have only desired dependencie…

### PR DESCRIPTION
**Pull request by mexanick**
…s for ARM and include postinstall scripts

### CODE REVIEW

File: specTemplate.spec [hunk 1]
```diff
@@ -33,6 +35,9 @@ BuildRoot: %{_tmppath}/%{_packagename}-%{_version}-%{_release}-buildroot
 Prefix: %{_prefix}
 Requires: __requires_list__
 BuildRequires: __build_requires_list__
+%if %{_is_arm}
+AutoReqProv: no
```
Comment (jsturdy) [inline specTemplate.spec]: `AutoReq` instead of `AutoReqProv`?

File: specTemplate.spec [hunk 2]
```diff
@@ -17,6 +17,8 @@
 
 %define _unpackaged_files_terminate_build 0
 
+%define _is_arm __is_arm__
```
Comment (jsturdy) [inline specTemplate.spec]: Prefer (for flexibility)

```spec
# Do not check any files in lib/arm for requires
%global __requires_exclude_from ^%{_prefix}/lib/arm/.*$
%global __requires_exclude_from ^%{_prefix}/lib/arm/*.*$
# Do not check any files in bin/arm for requires
%global __requires_exclude_from ^%{_prefix}/bin/arm/.*$
%global __requires_exclude_from ^%{_prefix}/bin/arm/*.*$
# Do not check .so files in an arm-specific library directory for provides
%global __provides_exclude_from ^%{_prefix}/lib/arm/*\\.so$
%global __provides_exclude_from ^%{_prefix}/lib/arm/*\.so$

%define add_arm_libs %( if [ -d 'lib/arm' ]; then echo "1" ; else echo "0"; fi )
%define is_arm  %( if [[ '__buildarch__' =~ "arm" ]]; then echo "1" ; else echo "0"; fi )
%define not_arm  %( if [[ ! '__buildarch__' =~ "arm" ]]; then echo "1" ; else echo "0"; fi )
```

File: mfRPMRules.mk [hunk 3]
```diff
@@ -89,6 +89,12 @@ _spec_update:
 	sed -i 's#__buildarch__#$(Arch)#' $(PackagePath)/rpm/$(PackageName).spec
 	sed -i 's#__requires_list__#$(REQUIRED_PACKAGE_LIST)#' $(PackagePath)/rpm/$(PackageName).spec
 	sed -i 's#__build_requires_list__#$(BUILD_REQUIRED_PACKAGE_LIST)#' $(PackagePath)/rpm/$(PackageName).spec
+	sed -i 's#__is_arm__#$(IS_ARM)#' $(PackagePath)/rpm/$(PackageName).spec
+
+	if [ -e $(PackagePath)/scripts/postinstall.sh ]; then \
+		sed -i '\#\bpost\b#r $(PackagePath)/scripts/postinstall.sh' $(PackagePath)/rpm/$(PackageName).spec; \
```
Comment (jsturdy) [inline mfRPMRules.mk:92]: Does `scripts` get packaged?
Do these scriptlets get packaged (and if so, do they need to be)?
I'd prefer that the scriptlets live either in the base directory and get copied into `rpm` during `preprpm`, or live in `rpm`, and `cleanrpm` gets changed to not `rm -rf rpm`

**Review by jsturdy — CHANGES_REQUESTED** [review]

File: specTemplate.spec (hunk 1 above)
Comment (mexanick) [reply specTemplate.spec]: nope. This is the correct syntax

File: mfRPMRules.mk (hunk 3 above)
Comment (mexanick) [reply mfRPMRules.mk:92]: >Does scripts get packaged?

yes, according to generic rules

>Do these scriptlets get packaged (and if so, do they need to be)?

Yes, but I don't really see any harm in this. I guess they don't have to be packaged as their content is copied into `spec`.

File: specTemplate.spec (hunk 2 above)
Comment (mexanick) [reply specTemplate.spec]: I don't see how this will improve the matter if things...

File: specTemplate.spec (hunk 1 above)
Comment (jsturdy) [reply specTemplate.spec]: I know it is "a" correct syntax, my question was whether we need to disable the "provides" functionality, or *just* the "requires" functionality

File: specTemplate.spec (hunk 2 above)
Comment (jsturdy) [reply specTemplate.spec]: The spec macros eliminate the need to define a variable in the `mfRPMRules.mk` file that is never used there, and only passed to the spec file, and allow the packaging of an `arm` library in a `devel` package (needed)

The other macros make sure that these packaged `arm` libs don't cause a failed installation due to requirements/provision checking in the non-`arm` package

File: specTemplate.spec (hunk 1 above)
Comment (mexanick) [reply specTemplate.spec]: hmmm... for some reason the `AutoReq: no` was giving an error before, so I thought it has been deprecated. But not anymore...

File: specTemplate.spec (hunk 2 above)
Comment (mexanick) [reply specTemplate.spec]: Ups, missed it was _already_ there. Extra var removed, code updated

**Review by jsturdy — APPROVED** [review]
Think we're "good enough" for now, may require a few more tweaks as we hammer out the complete packaging and distribution

**Pull request merged**